src/embedded_flash/syscall.rs

```diff
@@ -210,6 +213,117 @@ fn find_slice<'a>(
     Err(StorageError::OutOfBounds)
 }
 
+struct PartitionSlice {
+    pub ptr: usize,
+    pub len: usize,
+}
+
+impl PartitionSlice {
+    pub fn contains(&self, address: usize) -> bool {
+        // We want to check the 2 following inequalities:
+        // (1) `ptr <= address`
+        // (2) `address <= ptr + len`
```

Comment:
True, and the reason this actually passed is because the tests in this file were never run (including existing tests). I added
```
#[cfg(any(test, not(feature = "std")))]
mod helper;
```
and moved all tested helpers there.

Reply:
Also, the problematic code is removed if we keep the `RangeInclusive`. If you like the self-implemented `Range`, consider it fixed.